components/ConvertForm/ManageConversion.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useCallback, useEffect } from 'react'
+import {
+  useOpenOrder,
+  useClaimOrder,
+  useApprove,
+  useAllowance,
+} from 'lib/web3-contracts'
+import { bigNum } from 'lib/utils'
+import ConvertSteps from 'components/ConvertSteps/ConvertSteps'
+import processing from './assets/loader.gif'
+
+function ManageConversion({ toAnj, fromAmount, toAmount, handleReturnHome }) {
+  const openOrder = useOpenOrder()
+  const claimOrder = useClaimOrder()
+  const changeAllowance = useApprove()
+  const getAllowance = useAllowance()
+  const [conversionSteps, setConversionSteps] = useState()
+
+  // Interacting with the bonding curve involves 2, 3 or 4 transactions (depending on the direction and state of allowance):
+  // 1. Reset approval (If we're converting ANT -> ANJ, an allowance was previously set but abandoned)
+  // 2. Raise approval (If we're converting ANT -> ANJ, the current allowance is not high enough)
+  // 3. Open a buy order
+  // 4. Claim the order
+  const createConvertSteps = useCallback(async () => {
+    let buyOrderHash
+    let steps = []
+
+    // We first check for allowance if the direction is correct
+    if (toAnj) {
+      const allowance = await getAllowance()
+
+      // and if we need more, add a step to ask to approve more
+      if (allowance.lt(bigNum(fromAmount))) {
+        steps = [
```

Comment:
Just in case it hadn't occurred (no preference for either—being more immutable is good as well!), we can use `[].unshift()` and [].push()` since we already have an array declared at the top.

Reply:
I tend to favour immutability but on reflection this would probably be more readable!